### DISCUSSION TITLE
Aligning Dart’s gRPC web implementation

### DISCUSSION THREAD

**Discussion by bivens-dev** [grpc/grpc-web#1432] [General]
Hi there!

I saw in a talk last year that there was work underway to begin taking advantage of Dart’s [new JS interop support](https://medium.com/dartlang/dart-3-3-325bf2bf6c13) to begin a migration / alignment to use this version of the library so that things would remain in sync across both ecosystems.

I then saw something from a member of the Dart team saying that this work was abandoned but provided no details or specifics.

I just wanted to check in and see if that was actually the case and if so is anyone on the team able to provide any kind of information regarding:
- how come? 
- does gRPC-web continue to have a future in Dart 🤞 
- Was any kind of alternative proposed?

**Comment by sampajano** [comment]
@bivens-dev Hihi! Thanks for reaching out! 😃

You're definitely right about the intention to align grpc Dart to use gRPC-Web's implementations! But from what I understand it was never really put into action by anyone.. (the Dart team would be the main owner of this work)

> I then saw something from a member of the Dart team saying that this work was abandoned but provided no details or specifics.

> how come?

Could you refer to the source of where you get this information from? Maybe I can get a better idea then :)

> does gRPC-web continue to have a future in Dart 🤞

This definitely remain as an interest as far as gRPC-web team is concerned (more alignment means easier evolution of the platform / implementations). Although, since we don't own the Dart implementation, so it would be up to the Dart team to decide when this actually happens 😃

> Was any kind of alternative proposed?

Nothing as far as I'm aware of. I guess right now the interest remains low from Dart's side. Maybe as we evolve grpc-web and potentially add features in the upcoming year or so, this topic could potentially come up again :)

Hope that answers something :)

**Comment by mraleph** [comment]
The work was never started - it was simply considered (both by the Dart team and other internal users of Dart). But in the end nobody had either resources or motivation to do the work. Dart's gRPC-Web implementation works just fine right now (AFAIK - please correct me if I am wrong), so we have no reason to fix what is not broken. If this changes in the future for some reason - we will have to evaluate possible path forward.

**Reply by bivens-dev** [reply]
Thanks for the background and helping to clear up my confusion on this one.
